flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/JobResultEntry.java

```diff
@@ -22,41 +22,18 @@
 import org.apache.flink.util.Preconditions;
 
 /**
- * An entry in a {@link JobResultStore} that couples a completed {@link JobResult} to a state that
- * represents whether the resources of that JobResult have been finalized ({@link
- * JobResultState#CLEAN}) or have yet to be finalized ({@link JobResultState#DIRTY}).
+ * {@code JobResultEntry} is the entity managed by the {@link JobResultStore}. It collects
+ * information about a globally terminated job (e.g. {@link JobResult}).
  */
-public class JobResultEntry {
+public class JobResultEntry implements WithJobResult {
```

Comment:
Why do we need an extra `WithJobResult` interface?

Reply:
Initially, I though that it would be helpful for the file-based implementation (where we want to add the serializer version to persisted record). But looks like inheritance is good enough in that case. The interface is of no value. I'm going to delete this commit.

Reply:
I reverted the corresponding commit...